torch/masked/maskedtensor/_ops_refs.py

```diff
@@ -313,8 +313,7 @@ def _softmax(func, *args, **kwargs):
     _check_args_kwargs_length(args, kwargs, f"__torch_dispatch__, {func}", len_args=3, len_kwargs=0)
     data = _get_data(args[0])
     mask = _maybe_get_mask(args[0])
-    input_data = data.masked_fill(~mask, float("-inf"))
-    result_data = func(input_data, args[1], args[2])
+    result_data = torch.ops.aten._masked_softmax(data, ~mask, args[1], 1)
```

Comment:
1 is necessary because of https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/SoftMax.cpp#L143

Reply:
Based on the comments this maps to 
```
// If input is [B, H, T, T] and mask is [B, T]
  // we have special fast kernel
  // mask_type == 1 => mask_ is a src_key_padding_mask
```

But mask here should be of shape `[B, H, T, T]`. 

Based on the comment you're referencing It seems like the kernel on CPU checks the the mask is of shape `[T, T]`, which we don't have here.

Maybe `_masked_softmax` isn't equipped to handle these shapes on CPU or that check is wrong (then we should remove it).

Reply:
I think that check you're referencing should be removed if the dimension of the mask matches the dimension of the input. They're saying explicitly `[T, T]` is not supported, but our mask here should have dimension 4.

Reply:
Looking at [the CUDA kernel](https://github.com/pytorch/pytorch/blob/b182f0813519d9b153bf7acb4214c8e3f795866e/aten/src/ATen/native/cuda/SoftMax.cu#L964-L978) passing `1` as the mask type indicates a mask of shape `[B, T]` if it's 2-dim.

I'm worried someone might edit the CUDA kernel in the future and will raise an error due to this use site.

Maybe we should introduce a mask_type `2` that indicates the mask is of full dimension (which is our case).

Reply:
High level, the current implementation should still work because it’ll just have is_BxT_mask = 0 and then continue throughout normally since `mask.sizes() == input.sizes()`

But I agree it’s not the intended use case of `mask_type=1` and they could possibly add more error checks, so I’m happy with adding another `mask_type = 2`

Will create another PR to precede this one